tools/src/main/java/org/apache/kafka/tools/FeatureCommand.java

```diff
@@ -307,25 +294,24 @@ private static void update(String op, Admin admin, Map<String, FeatureUpdate> up
             }
         });
 
-        AtomicInteger numFailures = new AtomicInteger();
-        errors.keySet().forEach(feature -> {
-            short level = updates.get(feature).maxVersionLevel();
-            Optional<Throwable> maybeThrowable = errors.get(feature);
+        int numFailures = 0;
+        for (Map.Entry<String, Optional<Throwable>> entry : errors.entrySet()) {
```

Comment:
Also, in scala code, we traverse with sorted keys. Any reason we don't do it here?

Reply:
Spotbug is not happy when using keySet iterator and doing error.get(feature). And because we are not iterating on keyset, to traverse with sorted keys, we have to use TreeSet?

Reply:
Using TreeSet is good to me.